NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -1,140 +1,287 @@
+    sim_station.set_parameter(stnp.azimuth, azimuth)
+    sim_station.set_parameter(stnp.zenith, zenith)
+    sim_station.set_parameter(stnp.cr_energy, energy)
+    sim_station.set_magnetic_field_vector(magnetic_field_vector)
+    sim_station.set_parameter(stnp.cr_xmax, corsika['CoREAS'].attrs['DepthOfShowerMaximum'])
+    if 'highlevel' in corsika.keys():
+        sim_station.set_parameter(stnp.cr_energy_em, corsika["highlevel"].attrs["Eem"])
+    sim_station.set_is_cosmic_ray()
 
+    for j_obs, observer in enumerate(corsika['CoREAS']['observers'].values()):
+        obs_positions_geo = convert_obs_positions_to_nuradio_on_ground(observer, zenith, azimuth, magnetic_field_vector)
+        obs_positions_onsky = convert_obs_positions_to_vxB_vxvxB(observer, zenith, azimuth, magnetic_field_vector)
+        efield, efield_time = convert_obs_to_nuradio_efield(observer, zenith, azimuth, magnetic_field_vector)
+        electric_field = NuRadioReco.framework.electric_field.ElectricField(j_obs)
+        electric_field.set_trace(efield.T, sampling_rate)
+        electric_field.set_trace_start_time(efield_time[0])
+        electric_field.set_parameter(efp.ray_path_type, 'direct')
+        electric_field.set_parameter(efp.zenith, zenith)
+        electric_field.set_parameter(efp.azimuth, azimuth)
+        electric_field.set_position(obs_positions_geo)
+        electric_field.set_position_onsky(obs_positions_onsky)
+        sim_station.add_electric_field(electric_field)
+
+    evt = NuRadioReco.framework.event.Event(corsika['inputs'].attrs['RUNNR'], corsika['inputs'].attrs['EVTNR'])
+    stn = NuRadioReco.framework.station.Station(0) # set station id to 0
+    stn.set_sim_station(sim_station)
+    evt.set_station(stn)
+    evt.add_sim_shower(sim_shower)
+    corsika.close()
+    return evt
+
+def plot_footprint_onsky(sim_station, fig=None, ax=None):
+    """
+    plots the footprint of the observer positions in the (vxB, vxvxB) shower plane
 
-def make_sim_station(station_id, corsika, observer, channel_ids, weight=None):
+    Parameters
+    ----------
+    sim_station : sim station
+        simulated station object
     """
-    creates an NuRadioReco sim station from the (interpolated) observer object of the coreas hdf5 file
+    obs_positions = []
+    zz = []
+    for efield in sim_station.get_electric_fields():
+        obs_positions.append(efield.get_position_onsky())
+        zz.append(np.sum(efield[efp.signal_energy_fluence]))
+    obs_positions = np.array(obs_positions)
+    zz = np.array(zz)
+    if fig is None:
+        fig, ax = plt.subplots()
+    ax.set_aspect('equal')
+    sc = ax.scatter(obs_positions[:, 0], obs_positions[:, 1], c=zz, cmap=cm.gnuplot2_r, marker='o', edgecolors='k')
+    cbar = fig.colorbar(sc, ax=ax, orientation='vertical')
+    cbar.set_label('fluence [eV/m^2]')
+    ax.set_xlabel('vxB [m]')
+    ax.set_ylabel('vx(vxB) [m]')
+    return fig, ax
+
+
+def make_sim_station(station_id, corsika, weight=None):
```

Comment:
Should we keep this function around if it is deprecated? Maybe better to remove, then nobody can get confused

Reply:
@cg-laser , @fschlueter , opinions?

Reply:
Personally I like modularity, I would keep the function `make_sim_shower` but use it within `read_CORSIKA7`. With `make_sim_station` it is a bit different because we changed how we store the electric fields (several vs one sim station, correct?). Hence, it might be a good idea to remove this function. Or we keep it and rewrite it such that we can use in within `read_CORSIKA7` but also to create `sim_stations` for individual CoREAS observers.